integration-cli/docker_cli_links_test.go

```diff
@@ -169,50 +141,42 @@ func (s *DockerSuite) TestLinksUpdateOnRestart(c *check.C) {
 	if err != nil {
 		c.Fatal(err)
 	}
+	c.Assert(err, check.IsNil, check.Commentf("%v", err))
 	content, err := readContainerFileWithExec(id, "/etc/hosts")
-	if err != nil {
-		c.Fatal(err, string(content))
-	}
+	c.Assert(err, check.IsNil, check.Commentf("%v: %s", err, string(content)))
+
 	getIP := func(hosts []byte, hostname string) string {
 		re := regexp.MustCompile(fmt.Sprintf(`(\S*)\t%s`, regexp.QuoteMeta(hostname)))
 		matches := re.FindSubmatch(hosts)
-		if matches == nil {
-			c.Fatalf("Hostname %s have no matches in hosts", hostname)
-		}
+		c.Assert(matches, check.NotNil, check.Commentf("Hostname %s have no matches in hosts", hostname))
 		return string(matches[1])
 	}
-	if ip := getIP(content, "one"); ip != realIP {
-		c.Fatalf("For 'one' alias expected IP: %s, got: %s", realIP, ip)
-	}
-	if ip := getIP(content, "onetwo"); ip != realIP {
-		c.Fatalf("For 'onetwo' alias expected IP: %s, got: %s", realIP, ip)
-	}
+	ip := getIP(content, "one")
+	c.Assert(ip, check.Equals, realIP, check.Commentf("For 'one' alias expected IP: %s, got: %s", realIP, ip))
+
+	ip = getIP(content, "onetwo")
+	c.Assert(ip, check.Equals, realIP, check.Commentf("For 'onetwo' alias expected IP: %s, got: %s", realIP, ip))
```

Comment:
Same, I don't think we need the Commentf here. Assert will print the Obtained and Expected values.